File: scenarios/advanced/mobile/android-views/ComplexSearchInterface.java
package com.example.accessibilitybenchmark;

import android.app.AlertDialog;
import android.content.Context;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.*;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import java.util.List;

public class ComplexSearchInterface extends AppCompatActivity {
    
    private SearchView searchView;
    private RecyclerView filtersRecyclerView;
    private Spinner sortSpinner;
    private Spinner viewModeSpinner;
    private RecyclerView resultsRecyclerView;
    private ProgressBar loadingIndicator;
    private TextView emptyStateText;
    private Button clearFiltersButton;
    
    private List<SearchResult> searchResults = new ArrayList<>();
    private List<SearchResult> allResults = new ArrayList<>();
    private List<Filter> filters = new ArrayList<>();
    private List<String> selectedFilters = new ArrayList<>();
    private String currentSort = "relevance";
    private String currentViewMode = "list";
    private String selectedCategory = "All";
    private boolean isLoading = false;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_complex_search);
        
        initializeViews();
        setupFilters();
        setupSpinners();
        loadSampleData();
    }
    
    private void initializeViews() {
        searchView = findViewById(R.id.searchView);
        filtersRecyclerView = findViewById(R.id.filtersRecyclerView);
        sortSpinner = findViewById(R.id.sortSpinner);
        viewModeSpinner = findViewById(R.id.viewModeSpinner);
        resultsRecyclerView = findViewById(R.id.resultsRecyclerView);
        loadingIndicator = findViewById(R.id.loadingIndicator);
        emptyStateText = findViewById(R.id.emptyStateText);
        clearFiltersButton = findViewById(R.id.clearFiltersButton);
        
        // Setup RecyclerViews
        filtersRecyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false));
        resultsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        
        // Setup search
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                performSearch(query);
                return true;
            }
            
            @Override
            public boolean onQueryTextChange(String newText) {
                performSearch(newText);
                return true;
            }
        });
        
        clearFiltersButton.setOnClickListener(v -> clearAllFilters());
    }
    
    private void setupFilters() {
        filters.add(new Filter("recent", "Recent", "date"));
        filters.add(new Filter("large", "Large Files", "size"));
        filters.add(new Filter("images", "Images Only", "type"));
        filters.add(new Filter("videos", "Videos Only", "type"));
        filters.add(new Filter("documents", "Documents Only", "type"));
        filters.add(new Filter("free", "Free", "price"));
        filters.add(new Filter("premium", "Premium", "price"));
        
        FilterAdapter filterAdapter = new FilterAdapter(filters, this::onFilterToggle);
        filtersRecyclerView.setAdapter(filterAdapter);
    }
    
    private void setupSpinners() {
        // Sort spinner
        String[] sortOptions = {"relevance", "date", "name", "size", "type", "rating"};
        ArrayAdapter<String> sortAdapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, sortOptions);
        sortAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        sortSpinner.setAdapter(sortAdapter);
        
        sortSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                currentSort = sortOptions[position];
                performSearch(searchView.getQuery().toString());
            }
            
            @Override
            public void onNothingSelected(AdapterView<?> parent) {}
        });
        
        // View mode spinner
        String[] viewModes = {"list", "grid", "compact"};
        ArrayAdapter<String> viewModeAdapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, viewModes);
        viewModeAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        viewModeSpinner.setAdapter(viewModeAdapter);
        
        viewModeSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                currentViewMode = viewModes[position];
                // In a real implementation, this would change the view mode
            }
            
            @Override
            public void onNothingSelected(AdapterView<?> parent) {}
        });
    }
    
    private void loadSampleData() {
        allResults.add(new SearchResult(
            "1",
            "React Native Development Guide",
            "Comprehensive guide to building mobile apps with React Native",
            "document",
            "Documents",
            "2.5 MB",
            2.5,
            "2024-01-15",
            4.8,
            0.0,
            "📄",
            "https://example.com/react-native-guide"
        ));
        
        allResults.add(new SearchResult(
            "2",
            "Mobile UI Design Patterns",
            "Best practices for mobile user interface design",
            "image",
            "Images",
            "1.2 MB",
            1.2,
            "2024-01-14",
            4.6,
            29.99,
            "🖼️",
            "https://example.com/ui-patterns"
        ));
        
        allResults.add(new SearchResult(
            "3",
            "JavaScript Tutorial Series",
            "Complete JavaScript tutorial for beginners",
            "video",
            "Videos",
            "150 MB",
            150.0,
            "2024-01-13",
            4.9,
            49.99,
            "🎥",
            "https://example.com/js-tutorial"
        ));
        
        allResults.add(new SearchResult(
            "4",
            "Web Development News",
            "Latest news and updates in web development",
            "news",
            "News",
            "500 KB",
            0.5,
            "2024-01-12",
            4.3,
            0.0,
            "📰",
            "https://example.com/web-news"
        ));
        
        allResults.add(new SearchResult(
            "5",
            "E-commerce Product Catalog",
            "Complete product catalog for online store",
            "product",
            "Products",
            "5.8 MB",
            5.8,
            "2024-01-11",
            4.5,
            99.99,
            "🛍️",
            "https://example.com/product-catalog"
        ));
        
        searchResults = new ArrayList<>(allResults);
        updateResults();
    }
    
    private void performSearch(String query) {
        if (TextUtils.isEmpty(query.trim())) {
            searchResults = new ArrayList<>(allResults);
        } else {
            searchResults = new ArrayList<>();
            for (SearchResult result : allResults) {
                if (result.getTitle().toLowerCase().contains(query.toLowerCase()) ||
                    result.getDescription().toLowerCase().contains(query.toLowerCase()) ||
                    result.getCategory().toLowerCase().contains(query.toLowerCase())) {
                    searchResults.add(result);
                }
            }
        }
        
        applyFilters();
        updateResults();
    }
    
    private void applyFilters() {
        List<SearchResult> filteredResults = new ArrayList<>(searchResults);
        
        // Apply category filter
        if (!selectedCategory.equals("All")) {
            filteredResults.removeIf(result -> !result.getCategory().equals(selectedCategory));
        }
        
        // Apply selected filters
        for (String filterId : selectedFilters) {
            switch (filterId) {
                case "free":
                    filteredResults.removeIf(result -> result.getPrice() > 0);
                    break;
                case "premium":
                    filteredResults.removeIf(result -> result.getPrice() == 0);
                    break;
                case "large":
                    filteredResults.removeIf(result -> result.getSizeValue() <= 1.0);
                    break;
                case "images":
                    filteredResults.removeIf(result -> !result.getType().equals("image"));
                    break;
                case "videos":
                    filteredResults.removeIf(result -> !result.getType().equals("video"));
                    break;
                case "documents":
                    filteredResults.removeIf(result -> !result.getType().equals("document"));
                    break;
            }
        }
        
        searchResults = filteredResults;
    }
    
    private void onFilterToggle(String filterId) {
        if (selectedFilters.contains(filterId)) {
            selectedFilters.remove(filterId);
        } else {
            selectedFilters.add(filterId);
        }
        applyFilters();
        updateResults();
    }
    
    private void clearAllFilters() {
        selectedFilters.clear();
        selectedCategory = "All";
        searchView.setQuery("", false);
        searchResults = new ArrayList<>(allResults);
        updateResults();
    }
    
    private void updateResults() {
        if (searchResults.isEmpty()) {
            resultsRecyclerView.setVisibility(View.GONE);
            emptyStateText.setVisibility(View.VISIBLE);
            emptyStateText.setText("No results found\nTry adjusting your search terms or filters");
        } else {
            resultsRecyclerView.setVisibility(View.VISIBLE);
            emptyStateText.setVisibility(View.GONE);
            
            SearchResultAdapter adapter = new SearchResultAdapter(searchResults, this::onResultClick);
            resultsRecyclerView.setAdapter(adapter);
        }
    }
    
    private void onResultClick(SearchResult result) {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle(result.getTitle());
        
        StringBuilder message = new StringBuilder();
        message.append(result.getDescription()).append("\n\n");
        message.append("Type: ").append(result.getType()).append("\n");
        message.append("Category: ").append(result.getCategory()).append("\n");
        message.append("Size: ").append(result.getSize()).append("\n");
        message.append("Date: ").append(result.getDate()).append("\n");
        message.append("Rating: ").append(String.format("%.1f/5", result.getRating())).append("\n");
        message.append("Price: ").append(result.getPrice() > 0 ? 
            String.format("$%.2f", result.getPrice()) : "Free");
        
        builder.setMessage(message.toString());
        builder.setPositiveButton("Close", null);
        builder.setNegativeButton("Open", (dialog, which) -> {
            // Handle open action
        });
        builder.show();
    }
    
    // Filter Adapter
    private static class FilterAdapter extends RecyclerView.Adapter<FilterAdapter.ViewHolder> {
        private List<Filter> filters;
        private OnFilterClickListener listener;
        
        public interface OnFilterClickListener {
            void onFilterClick(String filterId);
        }
        
        public FilterAdapter(List<Filter> filters, OnFilterClickListener listener) {
            this.filters = filters;
            this.listener = listener;
        }
        
        @Override
        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_filter_chip, parent, false);
            return new ViewHolder(view);
        }
        
        @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            Filter filter = filters.get(position);
            holder.bind(filter, listener);
        }
        
        @Override
        public int getItemCount() {
            return filters.size();
        }
        
        static class ViewHolder extends RecyclerView.ViewHolder {
            private Button filterButton;
            
            public ViewHolder(View itemView) {
                super(itemView);
                filterButton = itemView.findViewById(R.id.filterButton);
            }
            
            public void bind(Filter filter, OnFilterClickListener listener) {
                filterButton.setText(filter.getName());
                filterButton.setOnClickListener(v -> listener.onFilterClick(filter.getId()));
            }
        }
    }
    
    // Search Result Adapter
    private static class SearchResultAdapter extends RecyclerView.Adapter<SearchResultAdapter.ViewHolder> {
        private List<SearchResult> results;
        private OnResultClickListener listener;
        
        public interface OnResultClickListener {
            void onResultClick(SearchResult result);
        }
        
        public SearchResultAdapter(List<SearchResult> results, OnResultClickListener listener) {
            this.results = results;
            this.listener = listener;
        }
        
        @Override
        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_search_result, parent, false);
            return new ViewHolder(view);
        }
        
        @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            SearchResult result = results.get(position);
            holder.bind(result, listener);
        }
        
        @Override
        public int getItemCount() {
            return results.size();
        }
        
        static class ViewHolder extends RecyclerView.ViewHolder {
            private TextView iconText;
            private TextView titleText;
            private TextView descriptionText;
            private TextView metadataText;
            private TextView priceText;
            
            public ViewHolder(View itemView) {
                super(itemView);
                iconText = itemView.findViewById(R.id.iconText);
                titleText = itemView.findViewById(R.id.titleText);
                descriptionText = itemView.findViewById(R.id.descriptionText);
                metadataText = itemView.findViewById(R.id.metadataText);
                priceText = itemView.findViewById(R.id.priceText);
            }
            
            public void bind(SearchResult result, OnResultClickListener listener) {
                iconText.setText(result.getImage());
                titleText.setText(result.getTitle());
                descriptionText.setText(result.getDescription());
                
                String metadata = result.getCategory() + " • " + result.getDate() + 
                                " • ⭐ " + String.format("%.1f", result.getRating()) + 
                                " • " + result.getSize();
                metadataText.setText(metadata);
                
                if (result.getPrice() > 0) {
                    priceText.setText("$" + String.format("%.2f", result.getPrice()));
                    priceText.setVisibility(View.VISIBLE);
                } else {
                    priceText.setText("Free");
                    priceText.setVisibility(View.VISIBLE);
                }
                
                itemView.setOnClickListener(v -> listener.onResultClick(result));
            }
        }
    }
    
    // Data classes
    public static class Filter {
        private String id;
        private String name;
        private String type;
        
        public Filter(String id, String name, String type) {
            this.id = id;
            this.name = name;
            this.type = type;
        }
        
        public String getId() { return id; }
        public String getName() { return name; }
        public String getType() { return type; }
    }
    
    public static class SearchResult {
        private String id;
        private String title;
        private String description;
        private String type;
        private String category;
        private String size;
        private double sizeValue;
        private String date;
        private double rating;
        private double price;
        private String image;
        private String url;
        
        public SearchResult(String id, String title, String description, String type, String category, 
                          String size, double sizeValue, String date, double rating, double price, 
                          String image, String url) {
            this.id = id;
            this.title = title;
            this.description = description;
            this.type = type;
            this.category = category;
            this.size = size;
            this.sizeValue = sizeValue;
            this.date = date;
            this.rating = rating;
            this.price = price;
            this.image = image;
            this.url = url;
        }
        
        // Getters
        public String getId() { return id; }
        public String getTitle() { return title; }
        public String getDescription() { return description; }
        public String getType() { return type; }
        public String getCategory() { return category; }
        public String getSize() { return size; }
        public double getSizeValue() { return sizeValue; }
        public String getDate() { return date; }
        public double getRating() { return rating; }
        public double getPrice() { return price; }
        public String getImage() { return image; }
        public String getUrl() { return url; }
    }
}

File: frameworks/mobile/flutter/lib/complex_search_interface.dart
import 'package:flutter/material.dart';

class ComplexSearchInterface extends StatefulWidget {
  const ComplexSearchInterface({Key? key}) : super(key: key);

  @override
  State<ComplexSearchInterface> createState() => _ComplexSearchInterfaceState();
}

class _ComplexSearchInterfaceState extends State<ComplexSearchInterface> {
  String searchQuery = '';
  List<Map<String, dynamic>> searchResults = [];
  bool isLoading = false;
  Set<String> selectedFilters = {};
  String sortOption = 'relevance';
  int currentPage = 1;
  int totalPages = 1;
  bool showFilters = false;
  List<String> searchHistory = [];
  bool showHistory = false;
  Map<String, dynamic>? selectedResult;
  String viewMode = 'list'; // list, grid, compact
  String selectedCategory = 'All';
  RangeValues priceRange = const RangeValues(0, 1000);
  DateTimeRange? dateRange;
  bool showAdvancedFilters = false;

  final List<String> categories = [
    'All', 'Documents', 'Images', 'Videos', 'Audio', 'Web Pages',
    'News', 'Products', 'People', 'Locations'
  ];

  final List<Map<String, dynamic>> sortOptions = [
    {'value': 'relevance', 'label': 'Relevance'},
    {'value': 'date', 'label': 'Date'},
    {'value': 'name', 'label': 'Name'},
    {'value': 'size', 'label': 'Size'},
    {'value': 'type', 'label': 'Type'},
    {'value': 'rating', 'label': 'Rating'},
  ];

  final List<Map<String, dynamic>> viewModes = [
    {'value': 'list', 'label': 'List', 'icon': Icons.list},
    {'value': 'grid', 'label': 'Grid', 'icon': Icons.grid_view},
    {'value': 'compact', 'label': 'Compact', 'icon': Icons.view_list},
  ];

  final List<Map<String, dynamic>> filters = [
    {'id': 'recent', 'name': 'Recent', 'type': 'date'},
    {'id': 'large', 'name': 'Large Files', 'type': 'size'},
    {'id': 'images', 'name': 'Images Only', 'type': 'type'},
    {'id': 'videos', 'name': 'Videos Only', 'type': 'type'},
    {'id': 'documents', 'name': 'Documents Only', 'type': 'type'},
    {'id': 'free', 'name': 'Free', 'type': 'price'},
    {'id': 'premium', 'name': 'Premium', 'type': 'price'},
  ];

  final List<Map<String, dynamic>> sampleResults = [
    {
      'id': '1',
      'title': 'React Native Development Guide',
      'description': 'Comprehensive guide to building mobile apps with React Native',
      'type': 'document',
      'category': 'Documents',
      'size': '2.5 MB',
      'date': '2024-01-15',
      'rating': 4.8,
      'price': 0,
      'image': '📄',
      'url': 'https://example.com/react-native-guide',
    },
    {
      'id': '2',
      'title': 'Mobile UI Design Patterns',
      'description': 'Best practices for mobile user interface design',
      'type': 'image',
      'category': 'Images',
      'size': '1.2 MB',
      'date': '2024-01-14',
      'rating': 4.6,
      'price': 29.99,
      'image': '🖼️',
      'url': 'https://example.com/ui-patterns',
    },
    {
      'id': '3',
      'title': 'JavaScript Tutorial Series',
      'description': 'Complete JavaScript tutorial for beginners',
      'type': 'video',
      'category': 'Videos',
      'size': '150 MB',
      'date': '2024-01-13',
      'rating': 4.9,
      'price': 49.99,
      'image': '🎥',
      'url': 'https://example.com/js-tutorial',
    },
    {
      'id': '4',
      'title': 'Web Development News',
      'description': 'Latest news and updates in web development',
      'type': 'news',
      'category': 'News',
      'size': '500 KB',
      'date': '2024-01-12',
      'rating': 4.3,
      'price': 0,
      'image': '📰',
      'url': 'https://example.com/web-news',
    },
    {
      'id': '5',
      'title': 'E-commerce Product Catalog',
      'description': 'Complete product catalog for online store',
      'type': 'product',
      'category': 'Products',
      'size': '5.8 MB',
      'date': '2024-01-11',
      'rating': 4.5,
      'price': 99.99,
      'image': '🛍️',
      'url': 'https://example.com/product-catalog',
    },
  ];

  @override
  void initState() {
    super.initState();
    if (searchQuery.isNotEmpty) {
      performSearch();
    }
  }

  void performSearch() async {
    setState(() {
      isLoading = true;
    });

    // Simulate API call
    await Future.delayed(const Duration(seconds: 1));

    List<Map<String, dynamic>> filteredResults = List.from(sampleResults);

    // Apply search query filter
    if (searchQuery.trim().isNotEmpty) {
      filteredResults = filteredResults.where((result) =>
          result['title'].toLowerCase().contains(searchQuery.toLowerCase()) ||
          result['description'].toLowerCase().contains(searchQuery.toLowerCase())
      ).toList();
    }

    // Apply category filter
    if (selectedCategory != 'All') {
      filteredResults = filteredResults.where((result) =>
          result['category'] == selectedCategory
      ).toList();
    }

    // Apply price range filter
    filteredResults = filteredResults.where((result) {
      final price = result['price'] as double;
      return price >= priceRange.start && price <= priceRange.end;
    }).toList();

    // Apply selected filters
    for (String filterId in selectedFilters) {
      final filter = filters.firstWhere((f) => f['id'] == filterId);
      switch (filter['type']) {
        case 'type':
          filteredResults = filteredResults.where((result) =>
              result['type'] == filterId.substring(0, filterId.length - 1)
          ).toList();
          break;
        case 'price':
          if (filterId == 'free') {
            filteredResults = filteredResults.where((result) => result['price'] == 0).toList();
          } else if (filterId == 'premium') {
            filteredResults = filteredResults.where((result) => result['price'] > 0).toList();
          }
          break;
        case 'size':
          if (filterId == 'large') {
            filteredResults = filteredResults.where((result) {
              final sizeStr = result['size'] as String;
              final size = double.tryParse(sizeStr.replaceAll(' MB', '')) ?? 0;
              return size > 1;
            }).toList();
          }
          break;
        case 'date':
          if (filterId == 'recent') {
            final recentDate = DateTime.now().subtract(const Duration(days: 7));
            filteredResults = filteredResults.where((result) {
              final resultDate = DateTime.parse(result['date']);
              return resultDate.isAfter(recentDate);
            }).toList();
          }
          break;
      }
    }

    // Apply sorting
    filteredResults.sort((a, b) {
      switch (sortOption) {
        case 'date':
          return DateTime.parse(b['date']).compareTo(DateTime.parse(a['date']));
        case 'name':
          return a['title'].compareTo(b['title']);
        case 'size':
          final aSize = double.tryParse(a['size'].toString().replaceAll(' MB', '')) ?? 0;
          final bSize = double.tryParse(b['size'].toString().replaceAll(' MB', '')) ?? 0;
          return bSize.compareTo(aSize);
        case 'type':
          return a['type'].compareTo(b['type']);
        case 'rating':
          return b['rating'].compareTo(a['rating']);
        default:
          return 0; // relevance - keep original order
      }
    });

    setState(() {
      searchResults = filteredResults;
      totalPages = (filteredResults.length / 10).ceil();
      currentPage = 1;
      isLoading = false;
    });
  }

  void handleSearch() {
    if (searchQuery.trim().isNotEmpty) {
      // Add to search history
      if (!searchHistory.contains(searchQuery)) {
        setState(() {
          searchHistory.insert(0, searchQuery);
          if (searchHistory.length > 10) {
            searchHistory = searchHistory.take(10).toList();
          }
        });
      }
      performSearch();
    }
  }

  void handleFilterToggle(String filterId) {
    setState(() {
      if (selectedFilters.contains(filterId)) {
        selectedFilters.remove(filterId);
      } else {
        selectedFilters.add(filterId);
      }
    });
  }

  void clearFilters() {
    setState(() {
      selectedFilters.clear();
      selectedCategory = 'All';
      priceRange = const RangeValues(0, 1000);
      dateRange = null;
    });
  }

  void handleResultPress(Map<String, dynamic> result) {
    setState(() {
      selectedResult = result;
    });
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: Text(result['title']),
        content: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(result['description']),
            const SizedBox(height: 8),
            Text('Type: ${result['type']}'),
            Text('Size: ${result['size']}'),
            Text('Date: ${result['date']}'),
            Text('Rating: ${result['rating']}/5'),
            Text('Price: \$${result['price']}'),
          ],
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context),
            child: const Text('Close'),
          ),
          TextButton(
            onPressed: () {
              Navigator.pop(context);
              // Handle open action
            },
            child: const Text('Open'),
          ),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[100],
      appBar: AppBar(
        title: const Text('Search'),
        backgroundColor: Colors.white,
        foregroundColor: Colors.black,
        elevation: 1,
      ),
      body: Column(
        children: [
          _buildSearchBar(),
          _buildFilters(),
          _buildSortAndView(),
          Expanded(
            child: _buildResults(),
          ),
        ],
      ),
    );
  }

  Widget _buildSearchBar() {
    return Container(
      padding: const EdgeInsets.all(16),
      color: Colors.white,
      child: Row(
        children: [
          Expanded(
            child: TextField(
              decoration: const InputDecoration(
                hintText: 'Search for anything...',
                prefixIcon: Icon(Icons.search),
                suffixIcon: Icon(Icons.clear),
                border: OutlineInputBorder(),
              ),
              onChanged: (value) => setState(() => searchQuery = value),
              onSubmitted: (_) => handleSearch(),
            ),
          ),
          const SizedBox(width: 12),
          ElevatedButton(
            onPressed: handleSearch,
            child: const Text('Search'),
          ),
        ],
      ),
    );
  }

  Widget _buildFilters() {
    return Container(
      padding: const EdgeInsets.symmetric(vertical: 8),
      color: Colors.white,
      child: SingleChildScrollView(
        scrollDirection: Axis.horizontal,
        child: Row(
          children: [
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 8),
              child: FilterChip(
                label: const Text('Filters'),
                selected: showFilters,
                onSelected: (selected) => setState(() => showFilters = selected),
                avatar: const Icon(Icons.filter_list, size: 16),
                selectedColor: Colors.blue[100],
                checkmarkColor: Colors.blue,
              ),
            ),
            ...filters.map((filter) => Padding(
              padding: const EdgeInsets.symmetric(horizontal: 4),
              child: FilterChip(
                label: Text(filter['name']),
                selected: selectedFilters.contains(filter['id']),
                onSelected: (selected) => handleFilterToggle(filter['id']),
                selectedColor: Colors.blue[100],
                checkmarkColor: Colors.blue,
              ),
            )),
          ],
        ),
      ),
    );
  }

  Widget _buildSortAndView() {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
      color: Colors.white,
      child: Row(
        children: [
          Expanded(
            child: Row(
              children: [
                const Text('Sort by: '),
                DropdownButton<String>(
                  value: sortOption,
                  onChanged: (value) => setState(() => sortOption = value!),
                  items: sortOptions.map((option) => DropdownMenuItem(
                    value: option['value'],
                    child: Text(option['label']),
                  )).toList(),
                ),
              ],
            ),
          ),
          Row(
            children: viewModes.map((mode) => IconButton(
              icon: Icon(
                mode['icon'],
                color: viewMode == mode['value'] ? Colors.blue : Colors.grey,
              ),
              onPressed: () => setState(() => viewMode = mode['value']),
            )).toList(),
          ),
        ],
      ),
    );
  }

  Widget _buildResults() {
    if (isLoading) {
      return const Center(
        child: CircularProgressIndicator(),
      );
    }

    if (searchResults.isEmpty && searchQuery.trim().isNotEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(Icons.search, size: 64, color: Colors.grey[400]),
            const SizedBox(height: 16),
            const Text(
              'No results found',
              style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
            ),
            const SizedBox(height: 8),
            const Text(
              'Try adjusting your search terms or filters',
              style: TextStyle(color: Colors.grey),
            ),
            const SizedBox(height: 16),
            ElevatedButton(
              onPressed: clearFilters,
              child: const Text('Clear all filters'),
            ),
          ],
        ),
      );
    }

    return ListView.builder(
      itemCount: searchResults.length,
      itemBuilder: (context, index) => _buildResultItem(searchResults[index]),
    );
  }

  Widget _buildResultItem(Map<String, dynamic> item) {
    return Card(
      margin: const EdgeInsets.symmetric(horizontal: 16, vertical: 4),
      child: ListTile(
        leading: CircleAvatar(
          backgroundColor: Colors.grey[200],
          child: Text(
            item['image'],
            style: const TextStyle(fontSize: 24),
          ),
        ),
        title: Text(
          item['title'],
          style: const TextStyle(fontWeight: FontWeight.w600),
        ),
        subtitle: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(item['description']),
            const SizedBox(height: 4),
            Row(
              children: [
                Icon(Icons.folder, size: 12, color: Colors.grey[600]),
                const SizedBox(width: 4),
                Text(item['category'], style: TextStyle(color: Colors.grey[600], fontSize: 12)),
                const SizedBox(width: 16),
                Icon(Icons.access_time, size: 12, color: Colors.grey[600]),
                const SizedBox(width: 4),
                Text(item['date'], style: TextStyle(color: Colors.grey[600], fontSize: 12)),
                const SizedBox(width: 16),
                Icon(Icons.star, size: 12, color: Colors.amber),
                const SizedBox(width: 4),
                Text(item['rating'].toString(), style: TextStyle(color: Colors.grey[600], fontSize: 12)),
                const SizedBox(width: 16),
                Icon(Icons.description, size: 12, color: Colors.grey[600]),
                const SizedBox(width: 4),
                Text(item['size'], style: TextStyle(color: Colors.grey[600], fontSize: 12)),
              ],
            ),
            if (item['price'] > 0) ...[
              const SizedBox(height: 4),
              Text(
                '\$${item['price']}',
                style: const TextStyle(
                  fontSize: 16,
                  fontWeight: FontWeight.bold,
                  color: Colors.blue,
                ),
              ),
            ],
          ],
        ),
        trailing: const Icon(Icons.chevron_right),
        onTap: () => handleResultPress(item),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[100],
      appBar: AppBar(
        title: const Text('Search'),
        backgroundColor: Colors.white,
        foregroundColor: Colors.black,
        elevation: 1,
      ),
      body: Column(
        children: [
          _buildSearchBar(),
          _buildFilters(),
          _buildSortAndView(),
          Expanded(
            child: _buildResults(),
          ),
        ],
      ),
      bottomSheet: showFilters ? _buildAdvancedFilters() : null,
    );
  }

  Widget _buildAdvancedFilters() {
    return Container(
      height: MediaQuery.of(context).size.height * 0.6,
      decoration: const BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.vertical(top: Radius.circular(20)),
      ),
      child: Column(
        children: [
          Container(
            padding: const EdgeInsets.all(16),
            decoration: const BoxDecoration(
              border: Border(bottom: BorderSide(color: Colors.grey, width: 0.5)),
            ),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                const Text(
                  'Filters',
                  style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
                ),
                Row(
                  children: [
                    TextButton(
                      onPressed: clearFilters,
                      child: const Text('Clear All'),
                    ),
                    IconButton(
                      onPressed: () => setState(() => showFilters = false),
                      icon: const Icon(Icons.close),
                    ),
                  ],
                ),
              ],
            ),
          ),
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.all(16),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  _buildFilterSection('Category', _buildCategoryFilters()),
                  const SizedBox(height: 24),
                  _buildFilterSection('Price Range', _buildPriceRangeFilter()),
                  const SizedBox(height: 24),
                  _buildFilterSection('Date Range', _buildDateRangeFilter()),
                ],
              ),
            ),
          ),
          Container(
            padding: const EdgeInsets.all(16),
            decoration: const BoxDecoration(
              border: Border(top: BorderSide(color: Colors.grey, width: 0.5)),
            ),
            child: SizedBox(
              width: double.infinity,
              child: ElevatedButton(
                onPressed: () {
                  setState(() => showFilters = false);
                  performSearch();
                },
                child: const Text('Apply Filters'),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildFilterSection(String title, Widget content) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          title,
          style: const TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
        ),
        const SizedBox(height: 12),
        content,
      ],
    );
  }

  Widget _buildCategoryFilters() {
    return Wrap(
      spacing: 8,
      runSpacing: 8,
      children: categories.map((category) => FilterChip(
        label: Text(category),
        selected: selectedCategory == category,
        onSelected: (selected) => setState(() => selectedCategory = category),
        selectedColor: Colors.blue[100],
        checkmarkColor: Colors.blue,
      )).toList(),
    );
  }

  Widget _buildPriceRangeFilter() {
    return Column(
      children: [
        RangeSlider(
          values: priceRange,
          min: 0,
          max: 1000,
          divisions: 20,
          labels: RangeLabels(
            '\$${priceRange.start.round()}',
            '\$${priceRange.end.round()}',
          ),
          onChanged: (values) => setState(() => priceRange = values),
        ),
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Text('\$${priceRange.start.round()}'),
            Text('\$${priceRange.end.round()}'),
          ],
        ),
      ],
    );
  }

  Widget _buildDateRangeFilter() {
    return Column(
      children: [
        ListTile(
          title: const Text('Start Date'),
          subtitle: Text(dateRange?.start.toString().split(' ')[0] ?? 'Select start date'),
          trailing: const Icon(Icons.calendar_today),
          onTap: () async {
            final date = await showDatePicker(
              context: context,
              initialDate: dateRange?.start ?? DateTime.now(),
              firstDate: DateTime(2020),
              lastDate: DateTime.now(),
            );
            if (date != null) {
              setState(() {
                dateRange = DateTimeRange(
                  start: date,
                  end: dateRange?.end ?? date.add(const Duration(days: 1)),
                );
              });
            }
          },
        ),
        ListTile(
          title: const Text('End Date'),
          subtitle: Text(dateRange?.end.toString().split(' ')[0] ?? 'Select end date'),
          trailing: const Icon(Icons.calendar_today),
          onTap: () async {
            final date = await showDatePicker(
              context: context,
              initialDate: dateRange?.end ?? DateTime.now(),
              firstDate: dateRange?.start ?? DateTime(2020),
              lastDate: DateTime.now(),
            );
            if (date != null) {
              setState(() {
                dateRange = DateTimeRange(
                  start: dateRange?.start ?? date.subtract(const Duration(days: 1)),
                  end: date,
                );
              });
            }
          },
        ),
      ],
    );
  }
}
